one/remote/aws.py

```diff
@@ -0,0 +1,134 @@
+"""
+This module is intended to provide a backend to download IBL data from AWS Buckets
+
+For example, without any credentials, to download a public file from the IBL public bucket:
+```
+import one.remote.aws as aws
+source = 'caches/unit_test/cache_info.json'
+destination = '/home/olivier/scratch/cache_info.json'
+aws.s3_download_file(source, destination)
+
+For a folder, the following
+```
+source = 'caches/unit_test'
+destination = '/home/olivier/scratch/caches/unit_test'
+local_files = aws.s3_download_public_folder(source, destination)
+"""
+
+
+from pathlib import Path
+import logging
+from tqdm import tqdm
+
+import boto3
+
+from botocore import UNSIGNED
+from botocore.config import Config
+from botocore.exceptions import NoCredentialsError, PartialCredentialsError, ClientError
+
+_logger = logging.getLogger(__name__)
+
+S3_BUCKET_IBL = 'ibl-brain-wide-map-public'
+REGION_NAME = 'us-east-1'
+
+
+def _callback_hook(t):
+    # https://gist.github.com/wy193777/e7607d12fad13459e8992d4f69b53586
+    # For example that uses actual file size:
+    # boto3.amazonaws.com/v1/documentation/api/latest/_modules/boto3/s3/transfer.html
+    def inner(bytes_amount):
+        t.update(bytes_amount)
+    return inner
+
+
+def get_aws_access_keys(one, repo_name='aws_cortexlab'):
+    """
+    Query alyx database to get credentials in the json field of an aws repository
+    :param one:
+    :param repo_name:
+    :return: dictionary of access keys to use in boto3, bucket name
+    """
+    repo_json = one.alyx.rest('data-repository', 'read', id=repo_name)['json']
+    bucket_name = repo_json['bucket_name']
+    session_keys = {
+        'aws_access_key_id': repo_json.get('Access key ID', None),
+        'aws_secret_access_key': repo_json.get('Secret access key', None)
+    }
+    return session_keys, bucket_name
+
+
+def get_s3_public():
+    session = boto3.Session(region_name=REGION_NAME)
+    s3 = session.resource('s3', config=Config(signature_version=UNSIGNED))
```

Comment:
@oliche Why is the signature version unsigned?

Reply:
It's for being able to instantiate a session without an account or anything and access public resources !